neuralprophet/forecaster.py

```diff
@@ -3359,7 +3359,7 @@ def conformal_predict(
             quantiles=self.config_train.quantiles,
```

Comment:
I'll be honest, the current structure here makes it basically unusable for the advanced user. That is because the advanced user has to basically call the Conformal class, get the noncon scores, use the noncon scores to get the q_hat, then put the q_hat into df. Those are all supposed to be behind-the-scenes in the `Conformal` method, regardless of the user.

I'd strongly prefer rather than all of that, you instead add one more param in the `conformal_predict()`, something like `show_conformal` etc. `False` is for the basic user (and set to default) and `True` is for the advanced user. If set to True, then the output of `df_forecast` is you currently have: only the QR columns. If set to False, then the output of `df_forecast` is what I currently have prior to this PR: columns with both QR and CP/CQR PIs.

Basically, there will be two workflows, one for basic and one for advanced. But that's ok as these workflows should be encapsulated within the `Conformal` class and `conformal_prediction()` will only pass whatever `Conformal` outputs.

With this, a couple more things:
1. `Conformal.predict()` should only output `df_forecast` and no need to output `noncon` scores, as the current setup.
2. `Conformal.plot()` should only be shown for advanced users. Something like:

```
if show_conformal and plotting_backend:
    c.plot(plotting_backend)
```

Reply:
Oke I understand, makes sense. I just want to avoid having the complete code duplicated, as this would mean doubled potential for bugs and double maintenance effort. Do you have an idea on how to avoid duplicated code?